### ISSUE TITLE
Enhance: Priority locks for database transactions

### ISSUE THREAD

**Issue by iltoga** [zoobc/zoobc-core#1424]
## Description
Since there is no way to prioritize database transactions at the moment, in case the node is overwhelmed by large number of transactions or other concurrent processes that lead to db writes, important and lenghtly processes such as processblock, processfork and downloadblockchain could be blocked by the first ones.

This is due to the fact that every time we execute a write operation or begin a db transaction, we lock other similar processes until the query has finished.

To avoid many low priority queries (such as writing new mempool transactions) to starve and block more important ones, we need to implement a priority sytem for database transactions. 

## Breakdown
- Adding this priorityLock implementation to our codebase (from https://github.com/platinummonkey/priority-lock)
- Refactor the relevant code (query/executor) to use priority locks instead of regular mutex ones

## Expected behavior
Even under heavy load, a node should not fall out of sync with its peers. to do that, we prioritize those processes that cannot stop due to high I/O operations
